frontend/src/api/tests/StreamApi.test.js

```diff
@@ -2,56 +2,163 @@ import axios from "axios";
 import Enzyme from "enzyme";
 import Adapter from "enzyme-adapter-react-16";
 import { afterEach, describe, expect, jest, it } from "@jest/globals";
-// import StreamApi itself
 import * as StreamApi from "../StreamApi";
+import StreamInfo from "../../model/StreamInfo";
+import { convertToDataObject } from "../../model/ConvertDataFormat";
 
 Enzyme.configure({ adapter: new Adapter() });
 
 jest.mock("axios");
 
-const mockSingleStream = {
-  streamId: 1,
-  streamName: "Test 1"
+const axiosAllStreamsResponse = [1, 2];
+const axiosGetFirstStreamResponse = {
+  id: 1,
+  inputChannel: {
+    id: 1,
+    channel: {
+      id: 1,
+      name: "Sample Receiver Channel 1",
+      port: 20002
+    },
+    decoder: {
+      serialNumber: "z7VBn9aK",
+      device: {
+        serialNumber: "z7VBn9aK",
+        publicIpAddress: "0:0:0:0:0:0:0:1",
+        privateIpAddress: "127.0.0.1",
+        displayName: "sample_receiver",
+        status: "online"
+      },
+      input: null
+    }
+  },
+  outputChannel: {
+    id: 2,
+    channel: {
+      id: 4,
+      name: "Sample Sender Channel 2",
+      port: 20001
+    },
+    encoder: {
+      serialNumber: "d7TxFn7o",
+      device: {
+        serialNumber: "d7TxFn7o",
+        publicIpAddress: "0:0:0:0:0:0:0:1",
+        privateIpAddress: "127.0.0.1",
+        displayName: "sample_sender",
+        status: "online"
+      },
+      output: null
+    }
+  },
+  isRendezvous: false
 };
-const mockStreams = [mockSingleStream, { streamId: 2, streamName: "Test 2" }];
 
-const respAllStreams = {
-  target: {
-    streams: mockStreams
-  }
-};
-const respSingleStream = {
-  target: {
-    stream: mockSingleStream
-  }
+const axiosGetSecondStreamResponse = {
```

Comment:
Do you actually need all this information? Since it's a mock call, can't you just make an object with the bare minimum properties?
getStream only uses stream id and the encoder & decoder. Thus you can simply have 
const dummyStream={
name:"test",
outputChannel: { encoder: {...}},
inputChannel: {decoder:{...}}
}

And you're mocking the return value anyways. You aren't using most of these fields

Reply:
I included all the fields returned by the backend in order to follow the principle that test data should reflect production data as closely as possible. Also, I suspect we'll use the full stream information when we create the detailed stream information page.